test/e2e/persistent_volume/persistent_volume_test.go

```diff
@@ -0,0 +1,349 @@
+
+var kClient k8s.KubeCtl
+var restClient yunikorn.RClient
+var dev = "dev-" + common.RandSeq(5)
+
+var _ = ginkgo.BeforeSuite(func() {
+	// Initializing kubectl client
+	kClient = k8s.KubeCtl{}
+	Ω(kClient.SetClient()).To(gomega.BeNil())
+
+	// Initializing rest client
+	restClient = yunikorn.RClient{}
+	Ω(restClient).NotTo(gomega.BeNil())
+
+	yunikorn.EnsureYuniKornConfigsPresent()
+
+	// Create namespace
+	ginkgo.By("Create namespace " + dev)
+	ns, err := kClient.CreateNamespace(dev, nil)
+	Ω(err).NotTo(HaveOccurred())
+	Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	// Clean up
+	err := kClient.DeletePVCs(dev)
+	Ω(err).NotTo(HaveOccurred())
+
+	err = kClient.DeletePVs(dev)
+	Ω(err).NotTo(HaveOccurred())
+
+	ginkgo.By("Tearing down namespace: " + dev)
+	err = kClient.TearDownNamespace(dev)
+	Ω(err).NotTo(HaveOccurred())
+})
+
+var _ = ginkgo.Describe("PersistentVolume", func() {
+	ginkgo.It("Verify_static_binding_of_local_pv", func() {
+		pvName := "local-pv-" + common.RandSeq(5)
+		conf := k8s.PvConfig{
+			Name:         pvName,
+			Capacity:     "1Gi",
+			AccessModes:  []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+			Source:       "Local",
+			Path:         "/tmp",
+			StorageClass: "standard",
+		}
+
+		pvObj, err := k8s.InitPersistentVolume(conf)
+		Ω(err).NotTo(HaveOccurred())
+		_, err = kClient.CreatePersistentVolume(pvObj)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(kClient.WaitForPersistentVolumeAvailable(pvName, time.Duration(60)*time.Second)).NotTo(HaveOccurred())
+		ginkgo.By("Create local type pv " + pvName)
+
+		pvcName := "pvc-" + common.RandSeq(5)
+		pvcConf := k8s.PvcConfig{
+			Name:       pvcName,
+			Capacity:   "1Gi",
+			VolumeName: pvName,
+		}
+
+		pvcObj, err := k8s.InitPersistentVolumeClaim(pvcConf)
+		Ω(err).NotTo(HaveOccurred())
+		_, err = kClient.CreatePersistentVolumeClaim(pvcObj, dev)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(kClient.WaitForPersistentVolumeClaimPresent(dev, pvcName, time.Duration(60)*time.Second)).NotTo(HaveOccurred())
+		ginkgo.By("Create pvc " + pvcName + ", which binds to " + pvName)
+
+		podName := "pod-" + common.RandSeq(5)
+		podConf := k8s.TestPodConfig{
+			Name:      podName,
+			Namespace: dev,
+			PvcName:   pvcName,
+		}
+
+		podObj, err := k8s.InitTestPod(podConf)
+		Ω(err).NotTo(HaveOccurred())
+		_, err = kClient.CreatePod(podObj, dev)
+		Ω(err).NotTo(HaveOccurred())
+		ginkgo.By("Create pod " + podName + ", which uses pvc " + pvcName)
+
+		err = kClient.WaitForPodRunning(dev, podName, time.Duration(60)*time.Second)
+		Ω(err).NotTo(HaveOccurred())
+		ginkgo.By("Check pod " + podName + " is successfully running")
+	})
+
+	ginkgo.It("Verify_dynamic_bindng_with_nfs_server", func() {
+		ginkgo.By("Start creating nfs provisioner.")
+
+		// Create nfs server and related rbac
+		saName := "nfs-service-account"
+		crName := "nfs-cluster-role"
+		crbName := "nfs-cluster-role-binding"
+		serverName := "nfs-provisioner"
+		scName := "nfs-sc"
```

Comment:
You can ignore the lintern by using `//nolint:gosec`. Run `make lint` locally to check.